### ISSUE TITLE
Nested structs mess up self_ty tracking

### ISSUE THREAD

**Issue by adetaylor** [google/autocxx#689]
(in context of the Chromium codebase)

`bindgen` gives us:

```rust
            extern "C" {
                #[bindgen_original_name("RenderFrameDeleted")]
                #[link_name = "\u{1}_ZN7content19WebContentsObserver18RenderFrameDeletedEPNS_15RenderFrameHostE"]
                pub fn WebContentsObserver_RenderFrameDeleted(
                    this: *mut ::std::os::raw::c_void,
                    render_frame_host: *mut root::content::RenderFrameHost,
                );
            }
```

which we parse, and is in our initial set of APIs:

```
      content::WebContentsObserver_RenderFrameDeleted (cpp=RenderFrameDeleted) (kind=Function)
```

but then this function is never seen again after the parsing phase. As such we generate:

```rust
pub trait WebContentsObserver_methods: WebContentsObserver_supers {}
```

which is about as useful as a chocolate teapot.

**Comment by adetaylor** [comment]
OK, tracked this down.

```
[2021-11-25T01:17:51Z INFO  autocxx_engine::conversion::analysis::fun] ADE: "RenderFrameDeleted" is not on allowlist, self ty is content::WebContentsObserver_MediaPlayerInfo
```

This is because we have structs nested inside `WebContentsObserver` and therefore the tracking of the `self_ty` is getting messed up.

**Comment by adetaylor** [comment]
Fixed by #694.
